Source/ProgressionSystemRuntime/Public/PSCWorldSubsystem.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Subsystems/WorldSubsystem.h"
+#include "PSCWorldSubsystem.generated.h"
+
+/**
+ * 
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSCWorldSubsystem : public UWorldSubsystem
+{
+	GENERATED_BODY()
+
+public:
+	/** Returns this Subsystem, is checked and wil crash if can't be obtained.*/
+	static UPSCWorldSubsystem& Get();
+	static UPSCWorldSubsystem& Get(const UObject& WorldContextObject);
+
+	/** Returns a progression System component reference */
+	FORCEINLINE TObjectPtr<class UProgressionSystemComponent> GetProgressionSystemComponent() const { return ProgressionSystemComponentInternal; }
```

Comment:
- TObjectPtr is needed only only for uproperties. In all other cases (functions, logic etc) just use raw pointer, so your TObjectPtr uproperty will be automatically converted: `FORCEINLINE class UProgressionSystemComponent* Get...`
- Make it `UFUNCTION(BlueprintPure, Category = "C++")', so it can be obtained in blueprints by designer (right now your property is not BlueprintReadWrite, so BP-Get is not available for this property, so expand at least this function)

Reply:
done

Reply:
There are still many wrong TObjectPtr Getters in Data Asset like `GetUnlockedProgressionIcon()`. Please go do Getters to be just raw pointers while only properties are TObjectPtr

Reply:
done